video/sweeper.go

```diff
@@ -0,0 +1,71 @@
+package video
+
+import (
+	"sort"
+	"sync"
+	"sync/atomic"
+)
+
+// A Counter is a thread-safe counter implementation
+type counter uint64
```

Comment:
fyi there's a whole library for nice atomic values: https://github.com/uber-go/atomic/blob/master/uint64.go#L59